packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -2197,7 +2247,10 @@ class LeaderLayer extends ContainerLayer {
     if (_link == value) {
       return;
     }
-    _link._leader = null;
+    if (attached) {
```

Comment:
wrote a test `switching layer link of an attached leader layer should not crash` for this change